packages/host/app/resources/search.ts

```diff
@@ -8,56 +9,64 @@ import { restartableTask } from 'ember-concurrency';
 import { Resource } from 'ember-resources';
 import flatMap from 'lodash/flatMap';
 
-import { TrackedMap } from 'tracked-built-ins';
+import { TrackedArray } from 'tracked-built-ins';
 
 import {
   subscribeToRealm,
   isCardCollectionDocument,
-  SingleCardDocument,
 } from '@cardstack/runtime-common';
 
 import type { Query } from '@cardstack/runtime-common/query';
 
-import type { CardDef } from 'https://cardstack.com/base/card-api';
+import { CardDef } from 'https://cardstack.com/base/card-api';
 
 import type { RealmEventContent } from 'https://cardstack.com/base/matrix-event';
 
-import { asURL } from '../services/store';
-
-import { type CardResource, getCard } from './card-resource';
-
 import type CardService from '../services/card-service';
 import type RealmServerService from '../services/realm-server';
+import type StoreService from '../services/store';
 
 const waiter = buildWaiter('search-resource:search-waiter');
 
 interface Args {
   named: {
     query: Query | undefined;
     realms: string[] | undefined;
-    isLive?: true;
+    isLive: boolean;
+    isAutoSaved: boolean;
     doWhileRefreshing?: (ready: Promise<void> | undefined) => Promise<void>;
+    storeService: StoreService;
```

Comment:
Why pass this in as an arg rather than inject it?

Reply:
nice catch, this was copy pasta